packages/node_modules/@webex/internal-plugin-presence/src/presence-worker.js

```diff
@@ -0,0 +1,276 @@
+
+import {debounce} from 'lodash';
+
+import {
+  FETCH_DELAY,
+  GROUNDSKEEPER_INTERVAL,
+  SUBSCRIPTION_DELAY,
+  UPDATE_PRESENCE_DELAY,
+  EXPIRED_PRESENCE_TIME,
+  PREMATURE_EXPIRATION_SUBSCRIPTION_TIME,
+  DEFAULT_SUBSCRIPTION_TTL,
+  APHELEIA_SUBSCRIPTION_UPDATE,
+  PRESENCE_UPDATE,
+  ENVELOPE_TYPE
+} from './constants';
+
+/**
+ * Presence Worker
+ * Manages fetches and subscriptions for presence
+ * @class
+ */
+export default class PresenceWorker {
+  /**
+   * Constructs a presence worker to execute and
+   * maintain tasks related to presence upkeep.
+   * @returns {undefined}
+   */
+  constructor() {
+    this.presences = {}; // current presence objects; updated time
+    this.watchers = {}; // counter of visible presence-required objects
+    this.fetchers = {}; // waiting to get presence object
+    this.flights = {}; // in flight to get presence object
+    this.campers = {}; // waiting to subscribe; updated time
+    this.subscribers = {}; // current subscriptions; expiration time
+  }
+
+  /**
+   * Connect to the mercury for presence and starts worker.
+   * @param {object} webex
+   * @returns {undefined}
+   */
+  initialize(webex) {
+    if (!webex || !webex.internal) {
+      throw new Error('Must initialize Presence Worker with webex!');
+    }
+
+    this.webex = webex;
+
+    const mercury = this.webex.internal.mercury.connected ?
+      Promise.resolve() : this.webex.internal.mercury.connect();
+
+    mercury.then(() => {
+      this.webex.internal.mercury.on(APHELEIA_SUBSCRIPTION_UPDATE,
+        this.subscriptionUpdate.bind(this));
+    });
+
+    setInterval(this.groundskeeper.bind(this), GROUNDSKEEPER_INTERVAL);
```

Comment:
Should this go inside the `then` block above?

Reply:
It doesn't need to. This sets up the cleaner for subs/presence which doesn't rely or affect mercury.